### PR TITLE
openstack: vol pop integration

### PR DESCRIPTION
**Pull request by bennyz**
- [x] Remove debug logs
- [x] Fix progress
- [x] Make sure errors are handled
- [x] Cleanup - remove temp snapshot, volume, image
- [x] Use storage profile for accessMode and volumeMode
- [x] Make it play nice with the oVirt changes

### CODE REVIEW

File: pkg/controller/plan/adapter/openstack/client.go
Comment (ahadas) [inline pkg/controller/plan/adapter/openstack/client.go:37]: should change later? if so, how about adding TODO?

File: pkg/controller/plan/kubevirt.go
Comment (ahadas) [inline pkg/controller/plan/kubevirt.go:573]: temporary?

**Review by ahadas — COMMENTED** [review]
took a quick look as it's at an early phase

File: cmd/populator-controller/populator-controller.go
Comment (bennyz) [inline cmd/populator-controller/populator-controller.go:61]: might be a blocking call, need to check

File: operator/roles/forkliftcontroller/templates/populator/deployment-populator-controller.yml.j2
Comment (bennyz) [inline operator/roles/forkliftcontroller/templates/populator/deployment-populator-controller.yml.j2:29]: @liranr23 FYI
had to remove it, because we need different ports for each controller

File: cmd/populator-controller/populator-controller.go
Comment (bennyz) [inline cmd/populator-controller/populator-controller.go:3]: @liranr23 please check this file

File: Makefile
Comment (mmartinv) [inline Makefile:233]: `hack/openstack-populator/Containerfile` is missing in the PR

Comment (bennyz) [reply Makefile:233]: thanks, added

File: pkg/controller/plan/adapter/openstack/builder.go
Comment (ahadas) [inline pkg/controller/plan/adapter/openstack/builder.go:249]: and also same as on the other PR, this shouldn't be necessarily (the Status section)

File: cmd/openstack-populator/openstack-populator.go
Comment (bennyz) [reply cmd/openstack-populator/openstack-populator.go:223]: you mean in forklift-controller? The progress isn't used to determine completion, it is considered completed when the PVC is bound

File: operator/config/manifests/bases/forklift-operator.clusterserviceversion.yaml
Comment (bennyz) [reply operator/config/manifests/bases/forklift-operator.clusterserviceversion.yaml:84]: Having the Volume part makes sense to me, but I don't feel too strongly about it, I'm fine with either

File: operator/config/samples/forklift_v1beta1_openstack_populator.yaml
Comment (bennyz) [reply operator/config/samples/forklift_v1beta1_openstack_populator.yaml:11]: It was required by mistake, but not anymore (for ovirt I think it still is)

File: pkg/controller/plan/kubevirt.go
Comment (bennyz) [reply pkg/controller/plan/kubevirt.go:1568]: I'll remove it, it will not be a part of this PR - it's a leftover from when I started dealing with image based VMs

File: cmd/openstack-populator/openstack-populator.go
Comment (ahadas) [reply cmd/openstack-populator/openstack-populator.go:225]: agree

File: operator/config/samples/forklift_v1beta1_openstack_populator.yaml
Comment (ahadas) [reply operator/config/samples/forklift_v1beta1_openstack_populator.yaml:11]: I think @liranr23 changed that since it was dropped from the samples, @liranr23 ?

File: operator/config/manifests/bases/forklift-operator.clusterserviceversion.yaml
Comment (ahadas) [reply operator/config/manifests/bases/forklift-operator.clusterserviceversion.yaml:84]: hehe, but both OvirtVolumePopulator and OpenstackVolumePopulator need 'ovp'.. so maybe for oVirt it can be 'vvp'..

Comment (bennyz) [reply operator/config/manifests/bases/forklift-operator.clusterserviceversion.yaml:84]: well, technically openstack is stylized OpenStack so it can be osvp 😄

File: pkg/controller/provider/container/openstack/client.go
Comment (bennyz) [reply pkg/controller/provider/container/openstack/client.go:50]: I will check this, I think having this uncommented causes an issue for the client in the controller because it's reused

File: pkg/controller/plan/adapter/openstack/builder.go
Comment (bennyz) [reply pkg/controller/plan/adapter/openstack/builder.go:280]: to indicate we don't want to reconcile again (it won't but I need to return something...)

File: pkg/controller/plan/adapter/openstack/builder.go
Comment (bennyz) [reply pkg/controller/plan/adapter/openstack/builder.go]: Added annotation and changed accordingly, hope I got it right

File: hack/openstack-populator/Containerfile
Comment (bennyz) [reply hack/openstack-populator/Containerfile:7]: I tried to copy vendor/ instead but I see it still downloads stuff so maybe I missed something

**Comment by bennyz** [comment]
@ahadas I think I addressed most stuff, some I need to look into some more, or copy from liran's PR

File: hack/openstack-populator/Containerfile
Comment (mnecas) [reply hack/openstack-populator/Containerfile:7]: yeah + I'm not sure how the d/s would like downloading the packages

File: cmd/populator-controller/populator-controller.go
Comment (liranr23) [reply cmd/populator-controller/populator-controller.go:3]: i will try after the ovirt populator PR. note that i changed some of the lib code. the populator will run on target namespace, the secret is mounted, etc.

File: hack/openstack-populator/Containerfile
Comment (mnecas) [inline hack/openstack-populator/Containerfile:10]: I think we should use the centos minimal version same as on the ovirt populator.
`FROM quay.io/centos/centos:stream9-minimal`

Comment (bennyz) [reply hack/openstack-populator/Containerfile:10]: In ovirt it was used because it needed some specific dependencies that weren't available, no @ahadas ?
That said, I don't mind changing it

File: hack/openstack-populator/Containerfile
Comment (bennyz) [reply hack/openstack-populator/Containerfile]: Changed to copying the vendor, not sure why it downloaded stuff previously

File: vendor/github.com/kubev2v/lib-volume-populator/populator-machinery/controller.go
Comment (bennyz) [inline vendor/github.com/kubev2v/lib-volume-populator/populator-machinery/controller.go:737]: revert before merging

**Comment by ahadas** [comment]
@bennyz please rebase

File: cmd/openstack-populator/openstack-populator.go
Comment (ahadas) [inline cmd/openstack-populator/openstack-populator.go:80]: we'll need a service account, right? better to change it as we do for ovirt-populator now

File: cmd/openstack-populator/openstack-populator.go
Comment (ahadas) [inline cmd/openstack-populator/openstack-populator.go:168]: please add TODO to implement the progress reporting

File: pkg/controller/plan/adapter/openstack/builder.go
Comment (ahadas) [inline pkg/controller/plan/adapter/openstack/builder.go:220]:
```suggestion
func (r *Builder) Secret(_ ref.Ref, in, secret *core.Secret) (err error) {
```

File: pkg/controller/plan/kubevirt.go
Comment (ahadas) [inline pkg/controller/plan/kubevirt.go:580]: isn't it problematic? (we won't get to useOvirtPopulator

Comment (bennyz) [reply pkg/controller/plan/kubevirt.go:580]: yeah...

Comment (ahadas) [reply pkg/controller/plan/kubevirt.go:580]: ack, I posted this before you fixed it

File: cmd/openstack-populator/openstack-populator.go
Comment (ahadas) [reply cmd/openstack-populator/openstack-populator.go]: actually better not that it at all if we don't need it - it was discussed in #12

File: pkg/controller/plan/adapter/openstack/builder.go
Comment (ahadas) [inline pkg/controller/plan/adapter/openstack/builder.go:219]: ah that comment is still here, why do we still need this?

File: vendor/github.com/kubev2v/lib-volume-populator/populator-machinery/controller.go
Comment (ahadas) [reply vendor/github.com/kubev2v/lib-volume-populator/populator-machinery/controller.go]: posted https://github.com/kubev2v/lib-volume-populator/pull/8

File: pkg/controller/plan/adapter/openstack/builder.go
Comment (bennyz) [reply pkg/controller/plan/adapter/openstack/builder.go:219]: Changed, but I think you previously commented in a different place because I copy-pasted it :)

**Review by ahadas — APPROVED** [review]
@bennyz can we drop the `preliminary` part from this PR before merging? I'll let you merge it with squash

**Pull request merged**
